asg3/SpellCheck.cpp

```diff
@@ -18,17 +18,34 @@ HashDict readDict(string dictFile) {
   ifstream data (dictFile.c_str());
   set<string> uniques;
   if (data.is_open()) {
-    while (getline(data,dictTemp)) {
-      int i =0;
-      while (dictTemp[i]) {
-        dictTemp[i] = (tolower(dictTemp[i]));
-        i++;
+    while(!data.eof()) {
+      // Parse in the entire word
+      string dictTemp = "";
+      char c;
+      data.get(c);
+      c = tolower(c);
+      while(isalpha(c) && !data.eof()) {
+        dictTemp += c;
+        data.get(c);
+        c = tolower(c);

```

Comment:
Using the different methods of reading word for checkFile and dictionary resulted in the words generating a different hash. My guess is that `getline()` left some sort of whitespace somewhere, but I did not experiment much to find out what was causing the different hash.